ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/remote/OciConnectorExecutor.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.engine.algorithms.remote;
+
+import static org.opensearch.ml.common.CommonValue.REMOTE_SERVICE_ERROR;
+import static org.opensearch.ml.common.connector.ConnectorProtocols.OCI_SIGV1;
+import static org.opensearch.ml.common.connector.OciConnector.OciClientAuthType;
+import static org.opensearch.ml.engine.algorithms.remote.ConnectorUtils.processOutput;
+
+import java.io.BufferedReader;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.Response;
+
+import org.opensearch.OpenSearchStatusException;
+import org.opensearch.common.util.TokenBucket;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.ml.common.connector.Connector;
+import org.opensearch.ml.common.connector.OciConnector;
+import org.opensearch.ml.common.exception.MLException;
+import org.opensearch.ml.common.input.MLInput;
+import org.opensearch.ml.common.output.model.ModelTensors;
+import org.opensearch.ml.engine.annotation.ConnectorExecutor;
+import org.opensearch.script.ScriptService;
+
+import com.oracle.bmc.Region;
+import com.oracle.bmc.auth.BasicAuthenticationDetailsProvider;
+import com.oracle.bmc.auth.InstancePrincipalsAuthenticationDetailsProvider;
+import com.oracle.bmc.auth.ResourcePrincipalAuthenticationDetailsProvider;
+import com.oracle.bmc.auth.SimpleAuthenticationDetailsProvider;
+import com.oracle.bmc.http.internal.RestClient;
+import com.oracle.bmc.http.internal.RestClientFactory;
+import com.oracle.bmc.http.internal.RestClientFactoryBuilder;
+import com.oracle.bmc.http.internal.WrappedInvocationBuilder;
+import com.oracle.bmc.http.signing.DefaultRequestSigner;
+import com.oracle.bmc.http.signing.RequestSigner;
+import com.oracle.bmc.requests.BmcRequest;
+
+import lombok.Getter;
+import lombok.Setter;
+import lombok.Synchronized;
+import lombok.extern.log4j.Log4j2;
+
+/**
+ * OciConnectorExecutor is responsible to call remote model from OCI services
+ */
+@Log4j2
+@ConnectorExecutor(OCI_SIGV1)
+public class OciConnectorExecutor implements RemoteConnectorExecutor {
+
+    @Getter
+    private final OciConnector connector;
+
+    @Setter
+    @Getter
+    private ScriptService scriptService;
+
+    @Setter
+    @Getter
+    private TokenBucket rateLimiter;
+
+    @Setter
+    @Getter
+    private Map<String, TokenBucket> userRateLimiterMap;
+
+    @Setter
+    @Getter
+    private org.opensearch.client.Client client;
```

Comment:
Nit: Not being used. Rate limiters are supported as part of ML models.

Reply:
It is for template design pattern
https://github.com/opensearch-project/ml-commons/blob/f5276ce9abbf9b57dfc0d1d5ba4195b939192749/ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/remote/RemoteConnectorExecutor.java#L129

We are doing the same thing here in AwsConnectorExecutor 
https://github.com/opensearch-project/ml-commons/blob/f5276ce9abbf9b57dfc0d1d5ba4195b939192749/ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/remote/AwsConnectorExecutor.java#L57

And for HttpConnectorExecutor
https://github.com/opensearch-project/ml-commons/blob/f5276ce9abbf9b57dfc0d1d5ba4195b939192749/ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/remote/HttpJsonConnectorExecutor.java#L56

Reply:
Yeah I guess this is the interface issue. For now just follow this pattern is fine I think.